pkg/controller/kafkachannel/reconciler.go

```diff
@@ -171,3 +187,32 @@ func (r *Reconciler) reconcile(ctx context.Context, channel *kafkav1beta1.KafkaC
 	// Return Success
 	return nil
 }
+
+// configMapObserver is the callback function that handles changes to our ConfigMap
+func (r *Reconciler) configMapObserver(configMap *corev1.ConfigMap) {
+	if configMap == nil {
+		r.logger.Warn("Nil ConfigMap passed to configMapObserver; ignoring")
+		return
+	}
+	if r == nil {
+		// This typically happens during startup
+		r.logger.Debug("Reconciler is nil during call to configMapObserver; ignoring changes")
+		return
+	}
+
+	// Though the new configmap could technically have changes to the eventing-kafka section as well as the sarama
+	// section, we currently do not do anything proactive based on configuration changes to those items.
+	// The only component in the controller that uses any of the fields after startup currently
+	// is the AdminClient, which simply uses the r.saramaConfig set here whenever necessary.
+	// This means that calling env.GetEnvironment and env.VerifyOverrides is not necessary now.  If
+	// those settings are needed in the future, the environment will also need to be re-parsed here.
+
+	// Load the Sarama settings from our configmap, ignoring the eventing-kafka result.
+	saramaConfig, _, err := commonconfig.LoadConfigFromMap(configMap)
+	if err != nil {
+		r.logger.Fatal("Failed To Load Eventing-Kafka Settings", zap.Error(err))
+	}
+
+	r.logger.Info("ConfigMap Changed; Updating Sarama Configuration")
+	r.saramaConfig = saramaConfig
+}
```

Comment:
is this the same function as in the channel?  can/should it be in common/config ?